citizen.ts

```diff
@@ -1,283 +1,138 @@
+  mkdirSync(temporaryContractsDir, { recursive: true });
 
-      return acc;
-    },
-    {} as ContractsByEpoch
+  const [, contractName] = rendezvousData.rendezvousContractId.split(".");
+  const rendezvousPath = join(
+    temporaryContractsDir,
+    `${contractName}-rendezvous.clar`
   );
-};
-
-/**
- * Deploys the contracts to the simnet in the correct order.
- * @param simnet The simnet instance.
- * @param contractsByEpoch The record of contracts by epoch.
- * @param getContractSourceFn The function to retrieve the contract source.
- */
-export const deployContracts = async (
-  simnet: Simnet,
-  contractsByEpoch: ContractsByEpoch,
-  manifestDir: string,
-  cacheDir: string,
-  getContractSourceFn: (
-    name: string,
-    sender: string,
-    props: ContractDeploymentProperties
-  ) => string
-): Promise<void> => {
-  for (const [epoch, contracts] of Object.entries(contractsByEpoch)) {
-    // Move to the next epoch and deploy the contracts in the correct order.
-    simnet.setEpoch(epoch as EpochString);
-    for (const contract of contracts.flatMap(Object.entries)) {
-      const [name, props]: [string, ContractDeploymentProperties] = contract;
-
-      // Resolve paths to absolute for proper comparison.
-      const absoluteContractPath = resolve(manifestDir, props.path);
-      const absoluteRequirementsPath = resolve(
-        manifestDir,
-        cacheDir,
-        "requirements"
-      );
-
-      // Check if contract is in requirements directory.
-      const isRequirement = absoluteContractPath.startsWith(
-        absoluteRequirementsPath
-      );
-
-      const sender = isRequirement
-        ? basename(props.path).split(".")[0]
-        : simnet.deployer;
+  writeFileSync(rendezvousPath, rendezvousData.rendezvousSourceCode);
 
-      const source = getContractSourceFn(name, sender, props);
+  radio.emit("logMessage", `Type-checking your unified Rendezvous contract...`);
+  const temporaryManifestPath = createTemporaryManifest(
+    parsedManifest,
+    manifestDir,
+    sutContractName,
+    rendezvousPath,
+    sessionId
+  );
 
-      simnet.deployContract(
-        name,
-        source,
-        { clarityVersion: props.clarity_version },
-        sender
-      );
+  try {
+    const simnet = await initSimnetSilently(temporaryManifestPath);
+    return simnet;
+  } finally {
+    unlinkSync(temporaryManifestPath);
+    rmSync(temporaryContractsDir, { recursive: true, force: true });
+
+    // Remove parent .rv-contracts directory if it exists and is empty.
+    const rvContractsParentDir = join(manifestDir, ".rv-contracts");
+    try {
+      if (existsSync(rvContractsParentDir)) {
+        const contents = readdirSync(rvContractsParentDir);
+        if (contents.length === 0) {
+          rmSync(rvContractsParentDir, { recursive: true, force: true });
+        }
+      }
+    } catch {
+      // Ignore errors when cleaning up parent directory.
```

Comment:
I think this needs some thinking. Users won't know if cleanup failed, leading to disk bloat.

At the very least:

```ts
} catch (error) {
  radio.emit("logMessage", 
    `Warning: Failed to cleanup temporary directory: ${error.message}`
  );
}
```